src/program/librapay.js

```diff
@@ -0,0 +1,234 @@
+// @flow
+
+import {
+  Account,
+  PublicKey,
+  Connection,
+  Transaction,
+  SystemProgram,
+} from '@solana/web3.js';
+import path from 'path';
+import fs from 'mz/fs';
+import * as lo from 'buffer-layout';
+
+import {newSystemAccountWithAirdrop} from '../util/new-system-account-with-airdrop';
+import * as InstructionData from './instruction_data';
+import {sendAndConfirmTransaction} from '../util/send-and-confirm-transaction';
+import {MoveLoader} from './move-loader';
+
+const sizeOfGenesisAccount = 5176;
+const sizeOfUserAccount = 2048; // TODO Generous estimate for most user accounts
+
+/**
+ * Load a new instance of a Move program on-chain.
+ * Returns the new program account.
+ */
+async function loadProgram(
+  connection: Connection,
+  path: string,
+): Promise<PublicKey> {
+  console.log(`Loading program: ${path}`);
+  const data = await fs.readFile(path);
+  const loaderAccount = await newSystemAccountWithAirdrop(connection, 100000);
+  return MoveLoader.load(connection, loaderAccount, data);
+}
+
+/**
+ * Create a new account owned by the Move loader program.
+ * Returns the newly created account.
+ */
+export async function createAccount(
+  connection: Connection,
+  payerAccount: Account,
+  size: number = sizeOfUserAccount,
+): Promise<Account> {
+  const newAccount = new Account();
+  const transaction = new Transaction();
+
+  transaction.add(
+    SystemProgram.createAccount(
+      payerAccount.publicKey,
+      newAccount.publicKey,
+      1,
+      size,
+      MoveLoader.programId,
+    ),
+  );
+
+  await sendAndConfirmTransaction(
+    `Send CreateGenesis command`,
+    connection,
+    transaction,
+    payerAccount,
+    newAccount,
+  );
+
+  return newAccount;
+}
+
+/**
+ * Create and populate a new Libra Genesis account.
+ * Returns the new account.
+ */
+export async function createGenesis(
+  connection: Connection,
+  payerAccount: Account,
+  amount: number,
+): Promise<Account> {
+  const genesisAccount = await createAccount(
+    connection,
+    payerAccount,
+    sizeOfGenesisAccount,
+  );
+
+  const transaction = new Transaction();
+  transaction.add({
+    keys: [
+      {
+        pubkey: genesisAccount.publicKey,
+        isSigner: true,
+        isDebitable: true,
+      },
+    ],
+    programId: MoveLoader.programId,
+    data: InstructionData.createGenesis(amount),
+  });
+
+  await sendAndConfirmTransaction(
+    `Send CreateGenesis Tx`,
+    connection,
+    transaction,
+    payerAccount,
+    genesisAccount,
+  );
+
+  return genesisAccount;
+}
+
+/**
+ * Mint tokens into a new Libra account.
+ * Returns the new account
+ */
+export async function mint(
+  connection: Connection,
+  payerAccount: Account,
+  genesisAccount: Account,
+  amount: number,
+): Promise<Account> {
+  const payeeAccount = await createAccount(connection, payerAccount);
+
+  const programPublicKey = await loadProgram(
+    connection,
+    path.join(__dirname, '../..', 'programs', 'mint_to_address.out'),
```

Comment:
Might as well break up `../..` into `'..', '..'` since you're already in a nice `path.join`